python/gym_ignition/randomizers/base/physics.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2020 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+import abc
+from gym_ignition import scenario_bindings as bindings
+
+
+class PhysicsRandomizer(abc.ABC):
+    """
+    Abstract class that provides the machinery for randomizing physics in a Ignition
+    Gazebo simulation.
+
+    Args:
+        randomize_after_rollouts_num: defines after many rollouts physics should be
+        randomized (i.e. the amount of times :py:meth:`gym.Env.reset` is called).
+    """
+
+    def __init__(self, randomize_after_rollouts_num: int = 0):
+
+        self._rollout_counter = randomize_after_rollouts_num
+        self.randomize_after_rollouts_num = randomize_after_rollouts_num
+
+    @abc.abstractmethod
+    def randomize_physics(self, world: bindings.World) -> None:
+        """
+        Method that insert and configures the physics of a world.
+
+        By default this method loads a plugin that uses DART with no randomizations.
+        Randomizing physics engine parameters or changing physics engine backend could be
+        done by redefining this method and passing it to
+        :py:class:`~gym_ignition.runtimes.gazebo_runtime.GazeboRuntime`.
+
+        Args:
+            world: A world object without physics.
+        """
+        pass
+
+    def seed_physics_randomizer(self, seed: int) -> None:
+        """
+        Seed the randomizer to ensure reproducibility.
+
+        Args:
+            seed: The seed number.
+        """
+        pass
+
+    def add_rollout_to_physics(self) -> None:
```

Comment:
This method name doesn't sound crystal clear to me.
What about `PhysicsRandomizer.increase_rollout_counter`?

Reply:
Yes that sounds better

Reply:
Anyway, this is still a rough edge among all the randomizations. I managed to integrate the physics in the randomized quantities but the logic is more convoluted than model and task. This is currently working fine but I'm not fully satisfied of how it works. Though, I couldn't find yet a cleaner way to do it, but you can expect that it will improved in the future (hopefully, by removing this counter).

Reply:
Update in 3a8e360

Reply:
Ok, souds good.